### PR TITLE
Record machine events

### PR DESCRIPTION
**Pull request by ingvagabund**
Record important error messages as events so they can be reported when describing a machine object.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -78,12 +78,16 @@ check: fmt vet lint test ## Check your code
...
-	$(DOCKER_CMD) go test -race -cover ./cmd/... ./pkg/...
+	$(DOCKER_CMD) go test -v -race -cover ./cmd/... ./pkg/...
 
 .PHONY: integration
 integration: ## Run integration test
 	$(DOCKER_CMD) go test -v sigs.k8s.io/cluster-api-provider-aws/test/integration
 
+.PHONY: build-e2e
+build-e2e:
+	go test -c -o bin/e2e.test sigs.k8s.io/cluster-api-provider-aws/test/machines
```
Comment (frobware) [inline Makefile]: Was wondering about the `-c` here; when/where is the binary run?

```
	-c
	    Compile the test binary to pkg.test but do not run it
	    (where pkg is the last element of the package's import path).
	    The file name can be changed with the -o flag.
```

**Comment by enxebre** [comment]
https://github.com/openshift/cluster-api-provider-aws/pull/109/commits/5666d6662c4e2ca04952c77e717e2377ba370be7 lgtm, let's make sure we agreed on 107 details

File: Makefile (hunk 1 above)
Comment (ingvagabund) [reply Makefile]: Inside the CI. Underlying Jenkins job runs `make build-e2e` and expects the binary under `bin/e2e.test`.

Comment (frobware) [reply Makefile]: Would be useful to add your answer as a comment for the next person. 👍

**Comment by bison** [comment]
This seem good. My one comment is mostly a style thing, which I hesitate to even bring up, but this makes a bunch of the lines extremely long. Like ~ 200 characters or more. I have a big 4k monitor and I can barely fit a whole line on the screen... I personally find that pretty hard to read.

A lot of the parameters are repeated. I feel like we could make some helpers / wrappers that made things a little more readable.

**Pull request merged**